iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/provisioning/setup/ProvisioningCommon.java

```diff
@@ -141,17 +141,29 @@ public static Collection inputsCommon(AttestationType attestationType)
         }
         else if (attestationType == AttestationType.TPM)
         {
-            return Arrays.asList(
-                    new Object[][]
-                            {
-                                    {ProvisioningDeviceClientTransportProtocol.HTTPS, attestationType},
-                                    {ProvisioningDeviceClientTransportProtocol.AMQPS, attestationType},
-                                    {ProvisioningDeviceClientTransportProtocol.AMQPS_WS, attestationType}
-
-                                    //MQTT/MQTT_WS does not support tpm attestation
-                                    //{ProvisioningDeviceClientTransportProtocol.MQTT, attestationType},
-                                    //{ProvisioningDeviceClientTransportProtocol.MQTT_WS, attestationType},
-                            });
+            if (!isPullRequest)
+            {
+                //TODO TPM tests are flakey, so only run them for CI and nightly builds
+                return Arrays.asList(
+                        new Object[][]
+                                {
+                                        {ProvisioningDeviceClientTransportProtocol.HTTPS, attestationType},
```

Comment:
So was this not referenced earlier?

Reply:
The TPM tests were being run previously. They were just returned as inputs above alongside x509 and symmetric key. This PR just splits the TPM tests into their own class so they can be run in serial while still running the x509 and symmetric key tests in parallel